### PR TITLE
feat: add category upload, convert HTML to Markdown, modularize services and utils, improve docs & tests

### PR DESCRIPTION
**Pull request by rmcnamara10**
## Summary
This PR introduces several improvements and new features:

- **Modularizes Confluence and Azure Storage services** for better maintainability and testability.
- **Adds recursive discovery** of all Confluence pages under a specified root, ensuring the full content hierarchy is uploaded to Azure Blob Storage
- **Enhances documentation** in the README with a detailed project overview, setup instructions, and environment variable guidance.
- **Refactors and expands unit tests** to cover the new modular structure and category upload logic, improving code coverage and reliability.

## Details
- Refactored core logic to separate Confluence and Azure Storage concerns into dedicated modules/services.
- Added support for uploading Confluence pages by category, with each page's metadata and content stored as a JSON file in the appropriate blob container.
- Updated README to provide clear setup, usage, and testing instructions for new and existing contributors.
- Reorganized and added new Jest unit tests for both Confluence and Azure Storage services.
- Cleaned up legacy/obsolete test files and ensured all tests pass with coverage reports generated.

## Motivation
These changes improve the maintainability, scalability, and clarity of the codebase, making it easier to extend and onboard new contributors. Category-based uploads enable more flexible downstream use of the data for AI and RAG scenarios.

---

**Please review the new service structure, test coverage, and updated documentation.**

Closes #<issue-number-if-applicable>

### CODE REVIEW

**Review by adamwozniak — COMMENTED** [review]
Smol comments nothing big. and one test failing:

ServiceCollectionExtensionsTests

AddSubscriptionsHandlers_RegistersHandlers() fails -

```
Message: 
Assert.Equal() Failure: Values differ
Expected: 12
Actual:   27
```

**Comment by rmcnamara10** [comment]
## Summary: Merge `convert-confluence-html` into `arch2.0-category-upload`

### Features

1. **Confluence HTML to Markdown Conversion**: Added `utils/htmlCleaner.js` that converts Confluence HTML content to Markdown, supporting headings, lists, tables, links, images, code, and more.
2. **Azure Storage Upload File**: `uploadPageData` in `azureStorageService.js` now uploads a JSON object containing `pageId`, `title`, `url`, and the new `cleanedText` (Markdown) field, instead of just HTML.
3. **Sample Output Generation**: Added `utils/sampleOutputWriter.js` to write sample HTML, Markdown, and JSON outputs for 5 pages when `ENABLE_SAMPLE_OUTPUTS` is enabled (for local debugging/inspection).

### Refactor

- Moved the `services` directory from `src/functions/services` to `src/services`
- Added a `src/utils` directory for HTML cleaning and writing sample output

### Testing

- Added unit tests in `htmlCleaner.test.js` to cover Markdown conversion for headings, lists, links, images, code, blockquotes, tables, and edge cases.

### Chores & Maintenance

- Removed unused HTTP trigger function, app setup files, and obsolete module exports.
- Updated `README.md` with the new `ENABLE_SAMPLE_OUTPUTS` environment variable and sample output feature.
- Updated `.gitignore` to exclude the `sample_output` directory

File: src/functions/cnfl_crawler.js [hunk 1]
```diff
@@ -1,101 +1,54 @@
...
-    } catch (error) {
-        return { 
-            error: {
-                pageId,
-                message: error.message
-            }
-        };
-    }
-}
+const CONF_ROOT_PAGE_ID = '64741277965'; // 'Architecture 2.0' root Confluence page
```
Comment (adamwozniak) [inline src/functions/cnfl_crawler.js:7]: maybe we can move it to env variables? it would be easier to change root page without code reploying

**Review by adamwozniak — APPROVED** [review]
approving with one small comment
